static/app/components/group/assignedTo.tsx

```diff
@@ -1,30 +1,132 @@
-import {useEffect} from 'react';
+import {useEffect, useState} from 'react';
 import styled from '@emotion/styled';
 
 import {fetchOrgMembers} from 'sentry/actionCreators/members';
+import {openCreateOwnershipRule} from 'sentry/actionCreators/modal';
+import Access from 'sentry/components/acl/access';
 import {
   AssigneeSelectorDropdown,
   AssigneeSelectorDropdownProps,
+  SuggestedAssignee,
 } from 'sentry/components/assigneeSelectorDropdown';
 import ActorAvatar from 'sentry/components/avatar/actorAvatar';
+import Button from 'sentry/components/button';
+import ButtonBar from 'sentry/components/buttonBar';
 import {AutoCompleteRoot} from 'sentry/components/dropdownAutoComplete/menu';
+import {
+  findMatchedRules,
+  Rules,
+} from 'sentry/components/group/suggestedOwners/findMatchedRules';
 import LoadingIndicator from 'sentry/components/loadingIndicator';
 import * as SidebarSection from 'sentry/components/sidebarSection';
-import {IconChevron, IconUser} from 'sentry/icons';
+import {IconAdd, IconChevron, IconSettings, IconUser} from 'sentry/icons';
 import {t} from 'sentry/locale';
 import MemberListStore from 'sentry/stores/memberListStore';
 import TeamStore from 'sentry/stores/teamStore';
 import space from 'sentry/styles/space';
-import type {Actor, Group, Project} from 'sentry/types';
+import type {Actor, Commit, Committer, Group, Project} from 'sentry/types';
+import type {Event} from 'sentry/types/event';
+import {defined} from 'sentry/utils';
 import useApi from 'sentry/utils/useApi';
+import useCommitters from 'sentry/utils/useCommitters';
 import useOrganization from 'sentry/utils/useOrganization';
 
 interface AssignedToProps {
   group: Group;
-  projectId: Project['id'];
+  project: Project;
   disableDropdown?: boolean;
+  event?: Event;
   onAssign?: AssigneeSelectorDropdownProps['onAssign'];
 }
+type IssueOwner = {
+  actor: Actor;
+  source: 'codeowners' | 'projectOwnership' | 'suspectCommit';
+  commits?: Commit[];
+  rules?: Array<[string, string]> | null;
+};
+type EventOwners = {
+  owners: Actor[];
+  rules: Rules;
+};
+
+function getSuggestedReason(owner: IssueOwner) {
```

Comment:
Is it possible to update this to return a SuggestedOwnerReason/update SuggestedOwnerReason? Does that make it cleaner/better? Not sure..

Reply:
we were using SuggestedOwnerReason before because we only cared what kind of suggestion "Codeowner". Now we want to display the full codeowner reason